### PR TITLE
Allow prefetch with IntermediateM2MField

### PR DESCRIPTION
**Pull request by Eg0ra**
Remove `M2MField` since `import_export.fields.Field` supports `m2m_add`
Improve `IntermediateManyToManyField` to make `prefetch_related` works with through model

### CODE REVIEW

**Review by TheSuperiorStanislav — APPROVED** [review]
Did this logic already have tests? If not let's add it

File: import_export_extensions/fields.py [hunk 1]
```diff
@@ -3,19 +3,40 @@
...
-    """Base M2M field provides faster related instances import."""
+class IntermediateManyToManyField(Field):
+    """Resource field for M2M with custom ``through`` model.
 
-    def __init__(self, *args, export_order=None, **kwargs):
-        """Save additional field params.
+    By default, ``django-import-export`` set up object attributes using
+    ``setattr(obj, attribute_name, value)``, where ``value`` is ``QuerySet``
+    of related model objects. But django forbid this when `ManyToManyField``
+    used with custom ``through`` model.
```
Comment (NikAzanov) [inline import_export_extensions/fields.py:12]: Please, re-check this and update docstring if needed

File: tests/test_models/test_fields.py [hunk 2]
```diff
@@ -45,10 +50,37 @@ def test_save_method(existing_artist: Artist, mocker: MockerFixture):
...
+def test_save_with_exception_during_full_clean(
+    existing_artist: models.Model,
+    band: Band,
+    mocker: pytest_mock.MockerFixture,
+):
+    """Check that the exception was handled during full_clean."""
+    # suggest widget returned following data
+    band_data = [
+        {"object": band, "properties": {"date_joined": "wrong_data_type"}},
+    ]
```
Comment (NikAzanov) [inline tests/test_models/test_fields.py:70]: Suggest removing comment and not to use separate variable, just

```python
mocker.patch(
    target=...,
    return_value=[
        {"object": band, "properties": {"date_joined": "wrong_data_type"}},
    ],
)
```

File: tests/test_models/test_fields.py [hunk 3]
```diff
@@ -45,10 +50,37 @@ def test_save_method(existing_artist: Artist, mocker: MockerFixture):
...
+        target=(
+            "import_export_extensions.fields.IntermediateManyToManyField.clean"
+        ),
+        return_value=band_data,
+    )
+
+    intermediate_field = IntermediateManyToManyField("bands")
+    with pytest.raises(
+        ValueError,
+        match=r"Column '.*':.*",
```
Comment (NikAzanov) [inline tests/test_models/test_fields.py:81]: Can we use more specified message here?

**Review by NikAzanov — APPROVED** [review]

File: tests/test_models/test_fields.py [hunk 4]
```diff
@@ -45,10 +50,50 @@ def test_save_method(existing_artist: Artist, mocker: MockerFixture):
...
+def test_save_readonly_field(existing_artist: Artist, band: Band):
     """Simple test to check that readonly field changes nothing."""
+    factories.MembershipFactory(artist=existing_artist, band=band)
     field = IntermediateManyToManyField(attribute="bands", readonly=True)
     field.save(existing_artist, {})
+    assert existing_artist.bands.all()
+
+
+def test_save_with_exception_during_full_clean(
+    existing_artist: models.Model,
```
Comment (OttoAndrey) [inline tests/test_models/test_fields.py:62]: And below for other tests
```suggestion
    existing_artist: Artist,
```

**Review by OttoAndrey — APPROVED** [review]

**Pull request merged**